core/client/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -319,7 +319,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
       // Canceled is considered a valid status and handled in the writer. We avoid creating a
       // CanceledException as an optimization.
       if (response.getMessage().asResponse().getStatus() != PStatus.CANCELED) {
-        response.unwrapException();
+        CommonUtils.unwrapResponse(response.getMessage().asResponse());
```

Comment:
I don't think we want to use extensions. How about having `Response` and `Type` fields first and second?